src/Uno.UI/UI/Xaml/Controls/SKCanvasElement.skia.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Numerics;
+using Windows.Foundation;
+using Windows.Graphics.Display;
+using Microsoft.UI.Composition;
+using Microsoft.UI.Xaml.Hosting;
+using SkiaSharp;
+using Uno.Disposables;
+
+namespace Microsoft.UI.Xaml.Controls;
+
+/// <summary>
+/// A wrapper around <see cref="SkiaVisual"/> that takes care of sizing, layouting and DPI.
+/// </summary>
+public abstract class SKCanvasElement : FrameworkElement
+{
+	private class SKCanvasVisual(SKCanvasElement owner, Compositor compositor) : SkiaVisual(compositor)
+	{
+		protected override void RenderOverride(SKCanvas canvas) => owner.RenderOverride(canvas, Size.ToSize());
+	}
+
+	private readonly SkiaVisual _skiaVisual;
+
+	protected SKCanvasElement()
+	{
+		_skiaVisual = new SKCanvasVisual(this, Visual.Compositor);
+		ElementCompositionPreview.SetElementChildVisual(this, _skiaVisual!);
+	}
+
+	public static DependencyProperty MirroredWhenRightToLeftProperty { get; } = DependencyProperty.Register(
+		nameof(MirroredWhenRightToLeft),
+		typeof(bool),
+		typeof(SKCanvasElement),
+		new FrameworkPropertyMetadata((dO, _) => ((SKCanvasElement)dO).RespectFlowDirectionChanged()));
+
+	/// <summary>
+	/// By default, SKCanvasElement will have the origin at the top-left of the drawing area with the normal directions increasing down and right.
+	/// If MirroredWhenRightToLeft is true, the drawing will be horizontally reflected when <see cref="SKCanvasElement.FlowDirection"/> is <see cref="FlowDirection.RightToLeft"/>.
+	/// </summary>
+	public bool MirroredWhenRightToLeft
```

Comment:
Not sure if we really need this API. I tend to avoid adding public APIs without enough motivation. We could probably postpone it unless it's clear that we need it.

Reply:
I agree, this is something that the user can do by hand, but it might be common enough and annoying enough to get right that we should provide it ourselves.